cmd/kms-rest/startcmd/start.go

```diff
@@ -588,17 +632,38 @@ func prepareKMSServiceCreator(keystoreService keystore.Service, cryptoService cr
 
 			return edv.NewStorageProvider(config)
 		}
-	} else {
-		kmsStorageResolver = func(string) (ariesstorage.Provider, error) {
+	default:
+		return func(string) (ariesstorage.Provider, error) {
 			return getKMSStorageProvider(params.keyManagerStorageParams)
 		}
 	}
+}
 
-	return kms.NewServiceCreator(&kms.Config{
-		KeystoreService:           keystoreService,
-		CryptoService:             cryptoService,
-		KeyManagerStorageResolver: kmsStorageResolver,
-	}), nil
+type keySecretLockResolver struct {
+	keyPath string
+}
+
+func (r *keySecretLockResolver) Resolve(_ *http.Request) (secretlock.Service, error) {
+	return getMasterKeySecretLock(r.keyPath)
+}
+
+func getSecretLockResolver(params *kmsRestParameters, tlsConfig *tls.Config) lock.Resolver {
+	switch {
+	case params.hubAuthURL != "":
```

Comment:
so we're not adding a "mode" flag?

Reply:
nope, at least for now

Reply:
I still believe that Key Server should not know about those roles as they are the level above Key Server functionality.